File: app/src/main/java/ua/hospes/rtm/domain/preferences/AbsPreferencesManager.java
package ua.hospes.rtm.domain.preferences;

import android.content.SharedPreferences;
import androidx.annotation.NonNull;

public abstract class AbsPreferencesManager {
    @NonNull
    public abstract SharedPreferences getPreferences();


    public int getInt(String key, int defaultValue) {
        return getPreferences().getInt(key, defaultValue);
    }

    public void putInt(String key, int value) {
        SharedPreferences.Editor editor = getPreferences().edit();
        editor.putInt(key, value);
        editor.apply();
    }

    public boolean putIntSync(String key, int value) {
        SharedPreferences.Editor editor = getPreferences().edit();
        editor.putInt(key, value);
        return editor.commit();
    }

    public float getFloat(String key, float defaultValue) {
        return getPreferences().getFloat(key, defaultValue);
    }

    public void putFloat(String key, float value) {
        SharedPreferences.Editor editor = getPreferences().edit();
        editor.putFloat(key, value);
        editor.apply();
    }

    public boolean putFloatSync(String key, float value) {
        SharedPreferences.Editor editor = getPreferences().edit();
        editor.putFloat(key, value);
        return editor.commit();
    }

    public long getLong(String key, long defaultValue) {
        return getPreferences().getLong(key, defaultValue);
    }

    public void putLong(String key, long value) {
        SharedPreferences.Editor editor = getPreferences().edit();
        editor.putLong(key, value);
        editor.apply();
    }

    public boolean putLongSync(String key, long value) {
        SharedPreferences.Editor editor = getPreferences().edit();
        editor.putLong(key, value);
        return editor.commit();
    }

    public String getString(String key, String defaultValue) {
        if (!getPreferences().contains(key)) {
            putString(key, defaultValue);
            return defaultValue;
        }
        return getPreferences().getString(key, defaultValue);
    }

    public void putString(String key, String value) {
        SharedPreferences.Editor editor = getPreferences().edit();
        editor.putString(key, value);
        editor.apply();
    }

    public boolean putStringSync(String key, String value) {
        SharedPreferences.Editor editor = getPreferences().edit();
        editor.putString(key, value);
        return editor.commit();
    }

    public boolean getBoolean(String key, boolean defaultValue) {
        return getPreferences().getBoolean(key, defaultValue);
    }

    public void putBoolean(String key, boolean value) {
        SharedPreferences.Editor editor = getPreferences().edit();
        editor.putBoolean(key, value);
        editor.apply();
    }

    public boolean putBooleanSync(String key, boolean value) {
        SharedPreferences.Editor editor = getPreferences().edit();
        editor.putBoolean(key, value);
        return editor.commit();
    }

    public void remove(String key) {
        SharedPreferences.Editor editor = getPreferences().edit();
        editor.remove(key);
        editor.apply();
    }
}

File: app/src/main/java/ua/hospes/rtm/domain/preferences/AbsPreferencesManager.kt
package ua.hospes.rtm.domain.preferences

import android.content.SharedPreferences

abstract class AbsPreferencesManager {
    abstract val preferences: SharedPreferences


    fun getInt(key: String, defaultValue: Int): Int {
        return preferences.getInt(key, defaultValue)
    }

    fun putInt(key: String, value: Int) {
        val editor = preferences.edit()
        editor.putInt(key, value)
        editor.apply()
    }

    fun putIntSync(key: String, value: Int): Boolean {
        val editor = preferences.edit()
        editor.putInt(key, value)
        return editor.commit()
    }

    fun getFloat(key: String, defaultValue: Float): Float {
        return preferences.getFloat(key, defaultValue)
    }

    fun putFloat(key: String, value: Float) {
        val editor = preferences.edit()
        editor.putFloat(key, value)
        editor.apply()
    }

    fun putFloatSync(key: String, value: Float): Boolean {
        val editor = preferences.edit()
        editor.putFloat(key, value)
        return editor.commit()
    }

    fun getLong(key: String, defaultValue: Long): Long {
        return preferences.getLong(key, defaultValue)
    }

    fun putLong(key: String, value: Long) {
        val editor = preferences.edit()
        editor.putLong(key, value)
        editor.apply()
    }

    fun putLongSync(key: String, value: Long): Boolean {
        val editor = preferences.edit()
        editor.putLong(key, value)
        return editor.commit()
    }

    fun getString(key: String, defaultValue: String): String? {
        if (!preferences.contains(key)) {
            putString(key, defaultValue)
            return defaultValue
        }
        return preferences.getString(key, defaultValue)
    }

    fun putString(key: String, value: String) {
        val editor = preferences.edit()
        editor.putString(key, value)
        editor.apply()
    }

    fun putStringSync(key: String, value: String): Boolean {
        val editor = preferences.edit()
        editor.putString(key, value)
        return editor.commit()
    }

    fun getBoolean(key: String, defaultValue: Boolean): Boolean {
        return preferences.getBoolean(key, defaultValue)
    }

    fun putBoolean(key: String, value: Boolean) {
        val editor = preferences.edit()
        editor.putBoolean(key, value)
        editor.apply()
    }

    fun putBooleanSync(key: String, value: Boolean): Boolean {
        val editor = preferences.edit()
        editor.putBoolean(key, value)
        return editor.commit()
    }

    fun remove(key: String) {
        val editor = preferences.edit()
        editor.remove(key)
        editor.apply()
    }
}